src/blob_format.h

```diff
@@ -253,10 +260,14 @@ class BlobFileMeta {
 //
 // The header is mean to be compatible with header of BlobDB blob files, except
 // we use a different magic number.
+//
+// If the version is 2, then the meta section includes a (possibly empty)
+// uncompression dictionary.
 struct BlobFileHeader {
   // The first 32bits from $(echo titandb/blob | sha1sum).
   static const uint32_t kHeaderMagicNumber = 0x2be0a614ul;
   static const uint32_t kVersion1 = 1;
+  static const uint32_t kVersion2 = 2;
```

Comment:
After a second thought, I think we should leave a 4 bytes padding in the file header, and use 1 bit as a flag to mean whether the file come with a dictionary. The rest of bits can leave room for future extension. I would like the have v2 mean "have extra flags in the header", not "have dictionary".

Reply:
Ok, I added a flags field. Right now it does nothing, but is serialized/deserialized when the version is 2 and complains if any unusual bits are set. I also didn't create an enum for the flags (since there's only one), but that would be an easy change if you'd prefer that. Let me know if I missed anything.